packages/react-charting/src/components/CommonComponents/ErrorBoundary.base.tsx

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import * as React from 'react';
+import { classNamesFunction, getId } from '@fluentui/react/lib/Utilities';
+import { IErrorBoundaryProps, IErrorBoundaryStyles } from './ErrorBoundary.types';
+import { ErrorBadgeIcon, ErrorIcon, StatusErrorFullIcon } from '@fluentui/react-icons-mdl2';
+
+export enum ErrorCodes {
+  NoData = 'No data',
+  GeneralError = 'General error',
+}
+
+interface IErrorBoundaryState {
+  hasError: boolean;
+  errorMsg: string;
```

Comment:
why do you need errorMsg as a state variable

Reply:
This is required so that we can set the aria label with the corresponding errorMsg as caught for unhandled exceptions in getDerivedStateFromError and componentDidCatch